TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -150,9 +150,25 @@ public override bool Visit(Section section)
             return true;
         }
 
-        public override bool Visit(Set setStatement)
+        public override bool Visit(Set set)
         {
-            SetStatementChecker.CheckStatement(setStatement);
+            if (set.StorageAreaWritesDataDefinition != null && set.CodeElement is SetStatementForConditions setConditions && setConditions.IsSendingValueFalse)
```

Comment:
Could this check be merged with the one added in `SetStatementChecker` to avoid iterating over conditions twice ?

Reply:
It is possible from a technical point of view.
But this check is related to Cobol and the one in `SetStatementChecker` is specific to TypeCobol.
As discussed it is better to keep them separate.